### PR TITLE
Implement serialize and deserialize for WasmModule

### PR DESCRIPTION
**Pull request by tqwewe**
WasmModule can be sent across processes on the same node by sharing the module ID via `push_module` and `take_module` host functions in PR https://github.com/lunatic-solutions/lunatic/pull/126

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -8,7 +9,7 @@ use thiserror::Error;
 /// them. This is especially true for calls that involve compiling raw binary data to WebAssembly
 /// modules. Because of this an opaque error ID is returned from host that can be transformed to
 /// a string.
-#[derive(Error)]
+#[derive(Error, Serialize, Deserialize)]
```
Comment (bkolobara) [inline src/error.rs:12]: It's not safe to serialise errors, they are also resources and bound the the running process.

File: src/module.rs [hunk 2]
```diff
@@ -26,6 +28,28 @@ impl Drop for WasmModule {
     }
 }
 
+impl Serialize for WasmModule {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        // TODO: Timeout info is not serialized
```
Comment (bkolobara) [inline src/module.rs:36]: I think this comment is just a leftover from a copy/paste.

**Comment by bkolobara** [comment]
Sorry, forgot to merge this. Could you just resolve the conflicts and I will do it?

**Comment by tqwewe** [comment]
I think this PR depends on the one in Lunatic https://github.com/lunatic-solutions/lunatic/pull/126

I did make a second branch which should avoid the performance change here: https://github.com/tqwewe/lunatic/tree/feat/shared-module-2
But I have yet to open a PR for it

**Pull request merged**